models/volunteer.py

```diff
@@ -0,0 +1,95 @@
+import datetime
+from models import camp
+
+
+class Volunteer:
+    """
+    A class used to represent a volunteer.
+    """
+
+    def __init__(self, firstname: str,
+                 lastname: str,
+                 phone: str,
+                 volunteercamp: list,
+                 availability = True):
+        """
+           Initialize a new volunteer.
+           :param firstname: the firstname of the volunteer
+           :param lastname: the lastname of the volunteer
+           :param phone: the phone number of the volunteer
+           :param volunteercamp: the camp that the volunteer belongs to
+           :param availability:  whether the volunteer is available to join a new emergency plan
+           :param creationdate: the date that this volunteer is created
+           """
+        currentdate = datetime.datetime.now()
+
+        self.name = self.__checkVolunteerName(firstname, lastname)
+        self.phone = self.__checkVolunteerPhone(phone)
+        self.camp = volunteercamp
+        self.availability = availability
+        self.creationdate = currentdate.date()
+
+
+    def __checkVolunteerName(self, firstname, lastname):
+        if len(firstname) <= 1:
+            raise self.InvalidFirstnameException(firstname)
+        elif len(lastname) <= 1:
+            raise self.InvalidLastnameException(lastname)
+        name = f"({firstname}+' '+{lastname})"
+        return name
```

Comment:
If the method also returns the name, maybe renaming it to `validate_name()` will be clearer. 
On the other hand, I cannot quite understand the name formatting. 
So we are storing names as e.g. "(Yunsy+' '+Yin)" ?
What is the motivation for this format?

Reply:
The check and return functions will be separated at the new branch "volunteerAccount". The reason we separated the fisrtname and the lastname is to avoid user confusion. It does seem unnecessary to combine them though, will update the setting at the new branch "volunteerAccount" as well.